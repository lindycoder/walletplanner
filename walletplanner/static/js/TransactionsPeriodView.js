import React from 'react';
import PropTypes from 'prop-types';
import TransactionBadge from "./TransactionBadge";


export default class TransactionsPeriodView extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div className={this.props.className}>
                <div className="transaction-view-date">{this.props.period.start.format("YYYY-MM-DD")}</div>
                <div className="transactions">
                    {this.props.transactions.map((transaction, transactionIndex) => (
                        <TransactionBadge key={transactionIndex} transaction={transaction}/>
                    ))}
                </div>
                <div className="transactions-total">
                    {formatAmount(this.props.transactions.reduce((total, e) => total + e.amount, 0))}
                </div>
            </div>
        )
    }
};

TransactionsPeriodView.propTypes = {
    period: PropTypes.any.isRequired,
    transactions: PropTypes.any.isRequired
};


function formatAmount(amount) {
    return `${(amount / 100).toFixed(2)}$`
}