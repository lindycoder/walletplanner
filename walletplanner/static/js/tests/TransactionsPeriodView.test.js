import React from 'react';
import {shallow} from 'enzyme';
import TransactionsPeriodView from '../TransactionsPeriodView';
import {moment} from '../utils';

describe('TransactionsPeriodView', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            period: moment.range(moment("2019-02-01"), moment("2019-02-15")),
            transactions: [
                {"amount": 1000, "date": moment("2019-02-06"), "description": "Current period 1", "category": null},
                {"amount": 1000, "date": moment("2019-02-08"), "description": "Current period 2", "category": null},
            ]
        };
        wrapper = shallow(<TransactionsPeriodView {...props} />);
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly amounts in the correct column', () => {
        expect(wrapper.find(".transaction-view-date").text()).toBe("2019-02-01");
        expect(wrapper.find(".transactions > TransactionBadge")).toHaveLength(2);

        expect(wrapper.find(".transactions-total").text()).toBe('20.00$');
    });
});
