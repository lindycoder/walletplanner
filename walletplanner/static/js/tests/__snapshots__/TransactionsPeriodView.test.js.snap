// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsPeriodView should render correctly 1`] = `
<div>
  <div
    className="transaction-view-date"
  >
    2019-02-01
  </div>
  <div
    className="transactions"
  >
    <TransactionBadge
      key="0"
      transaction={
        Object {
          "amount": 1000,
          "category": null,
          "date": "2019-02-06T00:00:00.000Z",
          "description": "Current period 1",
        }
      }
    />
    <TransactionBadge
      key="1"
      transaction={
        Object {
          "amount": 1000,
          "category": null,
          "date": "2019-02-08T00:00:00.000Z",
          "description": "Current period 2",
        }
      }
    />
  </div>
  <div
    className="transactions-total"
  >
    20.00$
  </div>
</div>
`;
